### PR TITLE
refactor: initial sendAmount logic for stardust

### PR DESCRIPTION
**Pull request by nicole-obrien**
## Summary
Please summarize your changes, describing __what__ they are and __why__ they were made.

### Changelog
```
Please write an accurate and concise changelog for your changes.
```

## Relevant Issues
Please list any related issues (e.g. bug, task).

## Type of Change
Please select any type below that is applicable to your changes, and delete those that are not.
- [ ] __Breaking__ - any change that would cause existing functionality to not work as expected
- [ ] __Chore__ - refactoring, build scripts or anything else that isn't user-facing
- [ ] __Docs__ - changes to the documentation
- [ ] __Fix__ - a change which fixes an issue
- [ ] __New__ - a change which implements a new feature
- [ ] __Update__ - a change which updates existing functionality

### Platforms
Please select any platforms where your changes have been tested.
- __Desktop__
	- [ ] MacOS
	- [ ] Linux
	- [ ] Windows
- __Mobile__
	- [ ] iOS
	- [ ] Android

### Instructions
Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

## Checklist
Please tick all of the following boxes that are relevant to your changes, and delete those that are not.
- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/actions/sendAmount.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export function sendAmount(receipientAddress: string, amount: number): Promise<TransactionReceipt[]> {
+    const account = get(selectedAccount)
+    const addressWithAmount: AddressWithAmount = { address: receipientAddress, amount: amount.toString() }
+    const transferOptions: TransferOptions = {
+        remainderValueStrategy: { strategy: 'ReuseAddress', value: null },
+        skipSync: true,
+        taggedDataPayload: {
+            type: 5,
+            tag: 'Sender',
+            data: 'Nicole waz here 2K22',
```
Comment (maxwellmattryan) [inline packages/shared/lib/core/wallet/actions/sendAmount.ts:14]: :eyes:

File: packages/shared/lib/core/wallet/actions/sendAmount.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { selectedAccount } from '@core/account'
+import { AddressWithAmount, TransactionReceipt, TransferOptions } from '@iota/wallet'
+import { get } from 'svelte/store'
+
+export function sendAmount(receipientAddress: string, amount: number): Promise<TransactionReceipt[]> {
+    const account = get(selectedAccount)
+    const addressWithAmount: AddressWithAmount = { address: receipientAddress, amount: amount.toString() }
+    const transferOptions: TransferOptions = {
+        remainderValueStrategy: { strategy: 'ReuseAddress', value: null },
```
Comment (maxwellmattryan) [inline packages/shared/lib/core/wallet/actions/sendAmount.ts:9]: Getting an error here:
```
ERROR in /home/maxwellmattryan/dev/iota/repos/firefly-pr/packages/shared/lib/core/wallet/actions/sendAmount.ts
[tsl] ERROR in /home/maxwellmattryan/dev/iota/repos/firefly-pr/packages/shared/lib/core/wallet/actions/sendAmount.ts(9,35)
      TS2322: Type '{ strategy: string; value: null; }' is not assignable to type 'RemainderValueStrategy'.
  Object literal may only specify known properties, and 'strategy' does not exist in type 'RemainderValueStrategy'.
```

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]

File: packages/shared/lib/core/profile-manager/interfaces/profile-manager.interface.ts [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
+import {
+    AccountId,
+    AccountSyncOptions,
+    ClientOptions,
+    CreateAccountPayload,
+    EventType,
+    WalletEvent,
+} from '@iota/wallet'
 
 export interface IProfileManager {
```
[the remark this answers is not in the document]
Comment (nicole-obrien) [reply packages/shared/lib/core/profile-manager/interfaces/profile-manager.interface.ts:13]: I think having the wrapper under this module is sensible and then specific actions will use them in the other modules.

File: packages/shared/lib/core/wallet/actions/sendAmount.ts (hunk 2 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/wallet/actions/sendAmount.ts:9]: The typing in the dependency is wrong waiting for David to update it.

File: packages/shared/lib/core/wallet/actions/sendAmount.ts (hunk 1 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/wallet/actions/sendAmount.ts:14]: lol removed

File: packages/shared/components/popups/SendConfirmationPopup.svelte [hunk 4]
```diff
@@ -23,8 +21,8 @@
 
         function _send(): void {
             return recipient.type === 'account'
-                ? sendInternalTransaction($selectedAccount.id, recipient.account.depositAddress, rawAmount, internal)
-                : sendExternalTransaction($selectedAccount.id, recipient.address, rawAmount)
+                ? void trySendAmount(recipient.account.depositAddress, rawAmount)
+                : void trySendAmount(recipient.address, rawAmount)
         }
```
Comment (MarkNerdi) [inline packages/shared/components/popups/SendConfirmationPopup.svelte:26]: `return void trySendAmount(recipient.type === 'account' ? recipient.account.depositAddress : recipient.address, rawAmount)`

File: packages/shared/lib/core/wallet/actions/sendAmount.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { selectedAccount } from '@core/account'
+import { AddressWithAmount, TransactionReceipt, TransferOptions } from '@iota/wallet'
+import { get } from 'svelte/store'
+
+export function sendAmount(receipientAddress: string, amount: number): Promise<TransactionReceipt[]> {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/wallet/actions/sendAmount.ts:5]: Spelling: `recipient`

File: packages/shared/lib/core/wallet/actions/trySendAmount.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
+import { localize } from '@core/i18n'
+import { isSoftwareProfile } from '@core/profile'
+import { showAppNotification } from '@lib/notifications'
+import { checkStronghold } from '@lib/stronghold'
+import { isTransferring } from '@lib/wallet'
+import { get } from 'svelte/store'
+import { sendAmount } from './sendAmount'
+
+export async function trySendAmount(receipientAddress: string, amount: number): Promise<void> {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/wallet/actions/trySendAmount.ts:9]: Spelling: `recipient`

File: packages/shared/lib/core/wallet/actions/sendAmount.ts (hunk 5 above)
Comment (MarkNerdi) [inline packages/shared/lib/core/wallet/actions/sendAmount.ts:5]: Is this function as an own action necessary?  I think we would always use the `trySendAmount` function in the app. If it is like that, I'd place it in the `trySendAmount.ts` without `export`ing it

File: packages/shared/lib/stronghold.ts [hunk 7]
```diff
@@ -36,12 +36,12 @@ export const checkStronghold = (callback: any): void => {
             } else {
                 callback()
             }
-        },
-        onError(err) {
+        })
+        .catch((err) => {
+            console.error(err)
```
Comment (MarkNerdi) [inline packages/shared/lib/stronghold.ts:41]: Do we accept `console.log`s on dev?

File: packages/shared/lib/core/wallet/actions/trySendAmount.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    const _send = async () => {
+        isTransferring.set(true)
+        try {
+            const transactionReceipts = await sendAmount(receipientAddress, amount)
```
Comment (MarkNerdi) [inline packages/shared/lib/core/wallet/actions/trySendAmount.ts:13]: we dont need that variable

File: packages/shared/lib/core/wallet/actions/trySendAmount.ts [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+            isTransferring.set(false)
+        } catch (err) {
+            console.error(err)
```
Comment (MarkNerdi) [inline packages/shared/lib/core/wallet/actions/trySendAmount.ts:16]: Do we accept `console.log`s on dev?

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
A few small comments on some code quality. 
Also: I get an error: `manager.isStrongholdPasswordAvailable is not a function at isStrongholdUnlocked` in the console

File: packages/shared/lib/core/wallet/actions/trySendAmount.ts (hunk 8 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/wallet/actions/trySendAmount.ts:13]: ahh yes, this was just to log it, and in the future we will do something with the actual response.

File: packages/shared/lib/core/wallet/actions/trySendAmount.ts (hunk 9 above)
Comment (nicole-obrien) [reply packages/shared/lib/core/wallet/actions/trySendAmount.ts:16]: no console logs, just console.error. We still need to agree on the error handling app wide.

File: packages/shared/lib/stronghold.ts (hunk 7 above)
Comment (nicole-obrien) [reply packages/shared/lib/stronghold.ts:41]: no console logs, just console.error. We still need to agree on the error handling app wide.

File: packages/shared/components/popups/SendConfirmationPopup.svelte (hunk 4 above)
Comment (Tuditi) [reply packages/shared/components/popups/SendConfirmationPopup.svelte:26]: Or: `const recipientAddress = recipient.type === 'account' ? recipient.account.depositAddress : recipient.address` and on a new line the trySendAmount function.

File: packages/shared/lib/stronghold.ts [hunk 10]
```diff
@@ -20,10 +21,9 @@ export const checkStronghold = (callback: any): void => {
         return
     }
 
-    api.getStrongholdStatus({
-        onSuccess(response: Event<StrongholdStatus>) {
-            const isLocked = response.payload.snapshot.status === 'Locked'
-            if (isLocked) {
+    isStrongholdUnlocked()
+        .then((response) => {
+            if (!response) {
```
Comment (Tuditi) [inline packages/shared/lib/stronghold.ts:26]: No callbacks please :)

**Review by Tuditi — APPROVED** [review]
If other comments are resolved and my concern is addressed, it's good to go from my part

**Pull request merged**